doc/nrf/ug_matter_overview_architecture.rst

```diff
@@ -0,0 +1,142 @@
+
+Matter architecture
+###################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Matter defines an application layer on top of the IPv6-based transport protocols.
+This allows for routing messages regardless of the underlying physical and link layers.
+
+.. figure:: images/matter_architecture.svg
+   :alt: Matter architecture overview
+
+   Matter architecture overview
+
+.. _ug_matter_architecture_overview:
+
+Matter stack architecture
+*************************
+
+The Matter application layer can be broken down into several main components, as shown on the following diagram.
+
+.. figure:: images/matter_components.svg
+   :alt: Matter stack overview
+
+   Matter stack overview
+
+At its lowest layer, the Matter stack interacts with the Transport layer.
+The payload flows down the protocol stack on the transmitting device and up the protocol stack on the receiving device.
+
+.. _ug_matter_architecture_overview_app:
+
+Application
+===========
+
+The Application layer defines the business logic for a given end product.
+For example, for a door lock application, the business logic could be opening and closing a specific model of the door bolt in reaction to a voice command from a specific virtual assistant technology.
+It could also define input from a specific PIN keyboard UI, reaction on a specific LED on the door bolt model, and so on.
+
+.. _ug_matter_architecture_overview_dm:
+
+Data Model
+==========
+
+The Data Model layer describes how to separate data into abstract logical blocks that help maintain the application in a comfortable manner.
+The layer operates regardless of the business logic, and so for example it is identical for each door lock type of device.
+
+.. figure:: images/matter_components_DM.svg
+   :alt: Data Model layer overview
+
+   Data Model layer overview
+
+Nodes
+   Each device is composed of one or more nodes, which are complete implementations of a Matter application functionality on a single stack.
+   Nodes are identifiable with a unique network address on a single network.
+   They can communicate directly with other nodes in the network.
+
+Endpoints
+   Each node is composed of one or more endpoints, which contain feature sets of single device functionalities.
+   For example, in a voice-controlled door lock device, one endpoint could include the feature set for operating the bolt, while the other endpoint could include the feature set for handling a temperature sensor.
+
+   .. note::
+      Endpoint 0 is always reserved for Matter's Utility Clusters.
+      This is the only mandatory endpoint for each Matter device.
+
+Clusters
+   Endpoints are composed of one or more clusters, which group attributes, commands and events that together make up each single feature in a set.
+   For example, in an endpoint for operating a door bolt, one cluster could group attributes for moving the bolt to open or closed position, while other could group attributes for controlling how fast the bolt would move.
```

Comment:
Generally door lock is not the easiest example to present variety of configurations as its all functionality is described by a single cluster. I'm not sure this how fast bolt will move is a real life attribute example for door lock, but maybe something like this could be better?
```suggestion
   For example, in an endpoint for operating a door bolt, one cluster could group attributes for moving the bolt to open or closed position, while other could group attributes for controlling alarms in case of opening the door in unauthorized way.
```